### PR TITLE
test: clean board

### PR DESCRIPTION
**Pull request by CatiaAntunes96**
Relates to:
- #1239 

## Proposed Changes

  - Add tests to ensure that clean-board function is returning everything as it should, with different arguments

This pull request closes:
- #1239

### CODE REVIEW

File: backend/src/modules/boards/utils/clean-boards.spec.ts [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+				card.createdAt = new Date(card.createdAt);
+				card.items.forEach((cardItem) => {
+					cardItem.createdAt = new Date(cardItem.createdAt);
+					cardItem.votes = hideVotes(
+						filterVotes(cardItem.votes as string[], String(createdByUser_1._id)),
+						String(createdByUser_1._id)
+					);
+				});
+			});
+		});
```
Comment (nunocaseiro) [inline backend/src/modules/boards/utils/clean-boards.spec.ts:334]: The votes from a card group should be taken into account

File: backend/src/modules/boards/utils/clean-boards.spec.ts [hunk 2]
```diff
@@ -0,0 +1,605 @@
...
+		board.columns = formatColumns(board.columns);
+
+		const boardResult = JSON.parse(JSON.stringify(board));
+
+		/* Format boardResult fields accordingly with what should be returned by the function:
+          - hide votes that are not from the user
+        */
+		boardResult.columns.forEach((column) => {
+			column.cards.forEach((card) => {
+				card.createdAt = new Date(card.createdAt);
```
Comment (nunocaseiro) [inline backend/src/modules/boards/utils/clean-boards.spec.ts:118]: The hide votes from the card group are missing

File: backend/src/modules/boards/utils/clean-boards.spec.ts [hunk 3]
```diff
@@ -0,0 +1,605 @@
...
+						String(createdByUser_1._id)
+					);
+				});
+			});
+		});
+
+		expect(cleanBoard(board, createdByUser_1._id)).toEqual(boardResult);
+	});
+
+	test('cleanBoard should return cards and votes hidden if hideCards and hidVotes are true', () => {
```
Comment (nunocaseiro) [inline backend/src/modules/boards/utils/clean-boards.spec.ts:339]:
```suggestion
	test('cleanBoard should return cards and votes hidden if hideCards and hideVotes are true', () => {
```

File: backend/src/modules/boards/utils/clean-boards.spec.ts [hunk 4]
```diff
@@ -0,0 +1,605 @@
...
+		const boardResult = JSON.parse(JSON.stringify(board));
+
+		/* Format boardResult fields accordingly with what should be returned by the function:
+        - hide text and replace user when card/cardItem if it is created by a user different form the arg userId and hideCards is true
+        - filter votes if hideVotes is true
+        - hide votes that are not from the user
+        */
+		boardResult.columns.forEach((column) => {
+			column.cards.forEach((card) => {
+				const createdByAsUser = card.createdBy as User;
```
Comment (nunocaseiro) [inline backend/src/modules/boards/utils/clean-boards.spec.ts:391]: The filter on card group votes is missing

**Review by nunocaseiro — CHANGES_REQUESTED** [review]

File: backend/src/modules/boards/utils/clean-boards.spec.ts [hunk 5]
```diff
@@ -0,0 +1,605 @@
...
+						comments: [],
+						items: [
+							CardItemFactory.create({
+								createdBy: cardUsers[0],
+								anonymous: false,
+								comments: [],
+								createdByTeam: null
+							})
+						]
+					},
```
[the remark this answers is not in the document]
Comment (CatiaAntunes96) [reply backend/src/modules/boards/utils/clean-boards.spec.ts:149]: Yes, the missing are automatically generated by the factory @nunocaseiro

**Review by nunocaseiro — APPROVED** [review]
LGTM 🚀 great job

**Pull request merged**